libmisc/myname.c

```diff
@@ -62,6 +62,7 @@
 		if ((NULL != pw) && (pw->pw_uid == ruid)) {
 			return pw;
 		}
+		pw_free (pw);
```

Comment:
This is not safe.  We could get here because pw == NULL, but then pw_free(pw) will not check for pw == NULL before doing free(pwent->pw_name);

Reply:
Good catch!